src/makeCalculator.js

```diff
@@ -37,7 +37,40 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  return {
+    add: (value, obj) => {
+      obj.result += value;
+
+      return obj;
+    },
+
+    subtract: (value, obj) => {
+      obj.result -= value;
+
+      return obj;
+    },
+    multiply: (value, obj) => {
+      obj.result *= value;
+
+      return obj;
+    },
+    divide: (value, obj) => {
+      obj.result /= value;
+
+      return obj;
+    },
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+    operate(callback, value) {
+      callback(value, this);
+
+      return this;
+    },
+    result: 0,
+  };
```

Comment:
You can optimize your code using "bind" or "apply"
(https://learn.javascript.ru/bind)

Reply:
This is valid, but rather optional comment

Reply:
> This is valid, but rather optional comment

I thought about implementing it, but was more focused on other tasks rather than trying to do it